File: LeetCode/Array/JS/Leetcode661-图片平滑器.js
var imageSmoother = function(img) {
    const n = img.length,m = img[0].length;
    const res = new Array(n).fill(0).map(()=>new Array(m).fill(0));
    let get = function(x,y) {
        let acc = 0,div = 0;
        for(let i = -1;i <= 1;i++) {
            for(let j = -1;j <= 1;j++){
                const nx = x + i,ny = y + j;
                if(nx < 0 || nx >= n || ny < 0 || ny >= m) continue;
                acc += img[nx][ny],div++; 
            }
        }
        return Math.floor(acc / div);
    }
    for(let i = 0;i < n;i++) {
        for(let j = 0;j < m;j++){
            res[i][j] = get(i,j);
        }
    }
    return res;
};

File: LeetCode/Array/Python/Leetcode661-图片平滑器.py
class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        n, m = len(img), len(img[0])
        res = [[0] * m for _ in range(n)]

        def get(x, y):
            acc, div = 0, 0
            for i in range(-1, 2):
                for j in range(-1, 2):
                    nx, ny = x + i, y + j
                    if nx < 0 or nx >= n or ny < 0 or ny >= m:
                        continue
                    acc += img[nx][ny]
                    div += 1
            return acc // div

        for i in range(n):
            for j in range(m):
                res[i][j] = get(i, j)
        return res
